src/settings.js

```diff
@@ -197,38 +197,48 @@ function getCliArgs(options = CLIOPTIONS, inputArgs = process.argv) {
  * @return {object}         - An object with errors and a boolean check
  */
 function checkCliInput(cliArgs, options = CLIOPTIONS) {
+	const result = {
+		pass: true,
+		errors: [],
+	};
+
 	const argDict = {};
-	Object.entries(options).map(([name, value]) => {
-		argDict[camelCase(name)] = name;
+	Object.entries(options).map(([key, value]) => {
+		argDict[camelCase(key)] = options[key];
 	});
 
-	Object.entries(cliArgs).map(([name, value]) => {
-		// console.log(name);
-		if (options[name]) {
-			// check type
-			// check arguments
+	Object.entries(cliArgs).map(([key, value]) => {
+		if (options[key]) {
+			console.log(`option found [${key}] for value [${value}]`);
+
+			// check types and check that the value matches at least one
+			if (
+				options[key].type === 'array' ? !Array.isArray(value) : typeof value !== options[key].type
+			) {
+				result.pass = false;
```

Comment:
I would also add a debug here:
```suggestion
				D.error(`Type mismatch found for "${color.yellow(key)}". Expected ${color.yellow(options[key].type)} but received ${color.yellow(typeof value)}`);
				result.pass = false;
```